app/controllers/observer_controller/other.rb

```diff
@@ -92,13 +95,12 @@ def set_export_status
     id    = params[:id].to_s
     type  = params[:type].to_s
     value = params[:value].to_s
-    model_class = type.camelize.safe_constantize
+    model_class = EXPORTABLE_MODELS.find { |m| m.name.downcase == type }
```

Comment:
Question: Should I get rid of the explanatory constant (and just plug the array in here?

Reply:
@pellaea: Is there way to get the db to supply the names of tables that have a `okay_to_export` column?
I took a first stab with the following (reformatted fro readability).
```
ActiveRecord::Base.connection.execute(%(
  SELECT 
  sys.columns.name AS ColumnName, 
  tables.name AS TableName 
  FROM 
  sys.columns 
  JOIN 
  sys.tables ON sys.columns.object_id = tables.object_id 
  WHERE 
  sys.columns.name = 'ok_for_export'
))
```
And got
>client.rb:148:in `_query': Mysql2::Error: SELECT command denied to user 'mo'@'localhost' for table 'columns' (ActiveRecord::StatementInvalid)
 /home/vagrant/.rvm/gems/ruby-3.0.4/gems/mysql2-0.5.4/lib/mysql2/client.rb:148:in `_query': SELECT command denied to user 'mo'@'localhost' for table 'columns' (Mysql2::Error)

Reply:
I don't think Nimmo will like any solution that involves explicit SQL.  He;'s nearly eradicated all SQL from the code by now.  Let's keep it that way.

It seems to me, though, you can probably work something out equivalent.  If there were a way to get a list of all ActiveRecord subclasses, for example, then you can filter those for ones that respond to okay_to_export, or have that attribute.  Since all of the models you care about here are subclasses of AbstractModel, maybe we could somehow use that to supply the list of all model classes.  Maybe provide a class method like AbstractModel.models_with_attribute(:okay_to_export).

Just some thoughts.